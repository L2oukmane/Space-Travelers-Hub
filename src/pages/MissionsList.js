import React from 'react';
import { useSelector } from 'react-redux';
import Mission from '../components/Mission';
import styles from './MissionsList.module.css';

const MissionsList = () => {
  const missions = useSelector((state) => state.missions);
  return (
    <ul className={styles.missionContainer}>
      <li className={styles.missionHeader}>
        <h3>Mission</h3>
        <h3>Description</h3>
        <h3>Status</h3>
      </li>
      {
        missions.length > 0
          ? missions.map((mission) => (
            <Mission
              key={mission.id}
              mission={mission}
            />
          ))
          : ''
      }
    </ul>
  );
};

export default MissionsList;
