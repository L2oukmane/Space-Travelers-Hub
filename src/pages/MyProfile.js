import React from 'react';
import { useSelector } from 'react-redux';
import ProfileMissions from '../components/ProfileMissions';
import RocketProfile from '../components/RocketProfile';
import styles from './MyProfile.module.css';

const MyProfile = () => {
  const missions = useSelector((state) => state.missions);
  const profileMission = missions.filter((mission) => mission.reserved === true);
  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  return (
    <div className={styles.container}>
      <div>
        <h2>My Missions</h2>
        {profileMission.length ? (
          <span>
            {profileMission.map((item) => (
              <ProfileMissions
                key={item.id}
                id={item.id}
                name={item.name}
              />
            ))}
          </span>
        ) : (
          <span className="nothing"><span>You have not joined any missions</span></span>
        )}
      </div>
      <div>
        <h2>My Rockets</h2>
        {reservedRockets.length > 0 ? (
          <span>
            {reservedRockets.map((rocket) => (
              <RocketProfile
                key={rocket.id}
                id={rocket.id}
                name={rocket.rocket_name}
              />
            ))}
          </span>
        ) : (
          <span className="nothing"><span>You have not reserved any rocket</span></span>
        )}
      </div>
    </div>
  );
};

export default MyProfile;
