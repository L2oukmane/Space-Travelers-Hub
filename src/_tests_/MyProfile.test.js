import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import MyProfile from '../pages/MyProfile';
import testStore from '../redux/configureStore';
import MissionsList from '../pages/MissionsList';

const mission = [
  {
    desc: 'Eutelsat S.A. is a European satellite operator',
    key: 'special',
    id: 1,
    name: 'defines the satellite',
    reservation: false,
  },
];

describe('testing the rendering of my profile page', () => {
  it('renders rocket and missions', () => {
    const { container } = render(
      <Provider store={testStore}>
        <MyProfile>
          <MissionsList key={mission[0].id} mission={mission[0]} />
        </MyProfile>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});