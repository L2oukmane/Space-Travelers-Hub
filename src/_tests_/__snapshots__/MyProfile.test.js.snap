// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of my profile page renders rocket and missions 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <h2>
        My Missions
      </h2>
      <span
        class="nothing"
      >
        <span>
          You have not joined any missions
        </span>
      </span>
    </div>
    <div>
      <h2>
        My Rockets
      </h2>
    </div>
  </div>
</div>
`;